File: 백준/Gold/1715. 카드 정렬하기/카드 정렬하기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.PriorityQueue;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int i = 0; i < n; i++) {
            pq.offer(Integer.parseInt(br.readLine()));
        }

        int total = 0;
        while (pq.size() > 1) {
            int sum = pq.poll() + pq.poll();
            total += sum;
            pq.offer(sum);
        }

        System.out.println(total);
    }
}

File: 백준/Gold/1715. 카드 정렬하기/카드 정렬하기.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.PriorityQueue

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()

    val pq = PriorityQueue<Int>()
    repeat(n) {
        pq.offer(br.readLine().toInt())
    }

    var total = 0
    while (pq.size > 1) {
        val sum = pq.poll() + pq.poll()
        total += sum
        pq.offer(sum)
    }

    println(total)
}
